front/src/reducers/osrdconf/types.ts

```diff
@@ -92,6 +92,23 @@ export type PathStep = PathItemLocation & {
   isInvalid?: boolean;
 };
 
-export type StdcmPathStep = PathStep & {
-  tolerances?: { before: number; after: number };
-} & ({ isVia: true; stopType: StdcmStopTypes } | { isVia: false; arrivalType: ArrivalTimeTypes });
+export type PathItemLocationWithMetadata<T> = Extract<PathItemLocation, T> & {
+  name: string;
+  coordinates: [number, number];
+};
+
+export type StdcmPathStep = {
+  id: string;
+  location?: PathItemLocation & {
```

Comment:
In stdcm, if I'm correct, a path step will always have an `uic`.

Also since recently, `PathItemLocation` has changed a bit to add the `track_reference` and we know distinguish `PathItemLocation` between a `TrackOffset` and an `OperationalPointReference`.
Should we start to use this more specific key word for our types ?

For example, `location` property could be renamed `reference` or something like that.

Anyway, by doing this, you could avoid all your `'uic' in step` checks across the code.

```suggestion
  location?: PathItemLocationWithMetadata<{ uic: number }> & {
```

Reply:
Yes you're right :+1: 
I was anticipating, but it's not a good thing